dashboard/engines/marketing/test/controllers/promotions_controller_test.rb

```diff
@@ -0,0 +1,70 @@
+require 'test_helper'
+require 'contentful'
+
+class PromotionsControllerTest < ActionDispatch::IntegrationTest
+  include Minitest::RSpecMocks
+
+  TestEntry = Struct.new(:content_type, :fields, keyword_init: true)
+  TestContentType = Struct.new(:id, keyword_init: true)
+
+  describe '#show' do
+    let(:locale) {'en-US'}
+    let(:entry_id) {SecureRandom.hex(10)}
+    let(:content_type) {TestContentType.new(id: 'teacherHomepageSidebar')}
+    let(:entry) do
+      TestEntry.new(
+        content_type: content_type,
+        fields: {
+          adPosition1: TestEntry.new(
+            fields: {
+              title: 'Ad 1',
+            }
+          ),
+          adPosition2: TestEntry.new(
+            fields: {
+              title: 'Ad 2',
+            }
+          ),
+        },
+      )
+    end
+    let(:expected_result) do
+      {
+        adPosition1: {
+          title: 'Ad 1'
+        },
+        adPosition2: {
+          title: 'Ad 2'
+        }
+      }.to_json
+    end
+
+    before do
+      Marketing::ContentfulClient.any_instance.expects(:entry).with(locale, entry_id).returns(entry)
+    end
+
+    it 'returns teacher sidebar json' do
+      get "/marketing/#{locale}/promotions/#{entry_id}", params: {locale: locale, id: entry_id}
+      assert_response :ok
+      assert_equal expected_result, @response.body
```

Comment:
If you’re committed to using Minitest's Spec syntax, it should look like this:
```suggestion
      must_respond_with :ok
      _(@response.body).must_equal expected_result
```

Alternatively, you can define the response subject for the response and test it like this:
```suggestion
      subject(:response) {@response.body.presence && JSON.parse(@response.body)}

      must_respond_with :ok
      # It’s better to compare hashes rather than strings for clearer and more precise test failure traces
      _response.must_equal JSON.parse(expected_result)
```